etl/steps/data/garden/lis/2023-01-18/luxembourg_income_study.py

```diff
@@ -0,0 +1,264 @@
+"""Load the three LIS meadow datasets and create one garden dataset, `luxembourg_income_study`."""
+
+import numpy as np
+import pandas as pd
+from owid.catalog import Dataset, Table
+from shared import add_metadata_vars
+from structlog import get_logger
+
+from etl.helpers import PathFinder
+
+log = get_logger()
+
+# Get paths and naming conventions for current step.
+paths = PathFinder(__file__)
+
+# Set relative and absolute poverty lines
+relative_povlines = [40, 50, 60]
+# Set equivalence scales
+equivalence_scales = ["eq", "pc"]
+
+# This function makes the table wide and modifies some columns before that
+# It is applied to the three LIS datasets
+def make_table_wide(df: pd.DataFrame, cols_to_wide: list) -> pd.DataFrame:
+
+    # Drop dataset variable, to not see it multiplied
+    df = df.drop(columns=["dataset"])
+
+    # Change names of equivalized variable, to create a distinguishable name
+    df["eq"] = df["eq"].replace({1: "eq", 0: "pc"})
+
+    # Create pivot table and join different levels of column
+    df = df.pivot(index=["country", "year"], columns=cols_to_wide).reset_index()
+    df.columns = [" ".join(col).strip().replace(" ", "_") for col in df.columns.values]
+
+    return df
+
+
+def run(dest_dir: str) -> None:
+    log.info("luxembourg_income_study.start")
+
+    #
+    # Load inputs.
+
+    ######################################################
+    # Key variables
```

Comment:
This function is really long. It would help readability if you split it to several functions, e.g. `load_keyvars_table`, `load_abs_poverty`, ... or something like that.

Reply:
Do you mean creating three functions in `shared` for each table or much more functions?

Reply:
It can be in the same module `luxembourg_income_study.py`. It's just that any function longer than 100 lines is not very readable and prone to errors. Splitting it into multiple functions should improve that and maybe even DRY the code. (I'm leaving how granular should it be up to you. You don't need to create tens of functions, but logically related code should be ideally wrapped in a function)

Reply:
Ah, OK, I asked because I didn't quite know what "very long" meant for a code.